### PR TITLE
fix: [PIE-4443]: Jenkins, amazonS3 triggers and cypress tests

### PR DESCRIPTION
**Pull request by satyam-harness**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: cypress/integration/72-triggers/AmazonS3Trigger.spec.ts [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    cy.contains('a', 'Triggers').click()
+    cy.wait(1000)
+    cy.contains('span', '+ New Trigger').click()
+    cy.get('section[data-cy="Artifact_Amazon S3"]').click()
+    cy.get('input[placeholder="Enter Name"]').clear().type('amazonS3Trigger')
+    cy.contains('p', '+ Select Artifact').click()
+    cy.get('.TableV2--cells').eq(0).click()
+    cy.contains('span', 'Select').click()
+    cy.wait(1000)
+    cy.get('input[placeholder="Select or Add bucket name"]').click()
```
Comment (udayvunnam) [inline cypress/integration/72-triggers/AmazonS3Trigger.spec.ts:56]: findByPlaceHolder() or findByRole with placeholder and can avoid explicit wait above

File: cypress/integration/72-triggers/AmazonS3Trigger.spec.ts [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+  })
+
+  it('testing amazonS3 trigger - when filePathRegex is runtime', () => {
+    cy.contains('a', 'Triggers').click()
+    cy.wait(1000)
+    cy.contains('span', '+ New Trigger').click()
+    cy.get('section[data-cy="Artifact_Amazon S3"]').click()
+    cy.get('input[placeholder="Enter Name"]').clear().type('amazonS3Trigger')
+    cy.contains('p', '+ Select Artifact').click()
+    cy.get('.TableV2--cells').eq(0).click()
```
Comment (udayvunnam) [inline cypress/integration/72-triggers/AmazonS3Trigger.spec.ts:53]: FYI cells have a role. we can use that

File: src/modules/72-triggers/pages/triggers/utils/TriggersListUtils.ts [hunk 3]
```diff
@@ -151,6 +155,17 @@ const triggerDrawerMap = (
           value: ENABLED_ARTIFACT_TYPES.Acr,
           iconName: ArtifactIconByType.Acr
         }
+        // Uncomment this to enable AmazonS3 and Jenkins triggers
```
Comment (udayvunnam) [inline src/modules/72-triggers/pages/triggers/utils/TriggersListUtils.ts:158]: Let's remove the commented code unless we can look back at this with ticket number

File: cypress/integration/72-triggers/AmazonS3Trigger.spec.ts [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+    }).as('s3bucketData')
+
+    visitTriggersPageWithAssertion()
+  })
+
+  it('testing amazonS3 trigger - when filePathRegex is runtime', () => {
+    cy.contains('a', 'Triggers').click()
+    cy.wait(1000)
+    cy.contains('span', '+ New Trigger').click()
+    cy.get('section[data-cy="Artifact_Amazon S3"]').click()
```
[the remark this answers is not in the document]
Comment (satyam-harness) [reply cypress/integration/72-triggers/AmazonS3Trigger.spec.ts:50]: we discussed this is not an issue

File: cypress/integration/72-triggers/AmazonS3Trigger.spec.ts (hunk 2 above)
Comment (satyam-harness) [reply cypress/integration/72-triggers/AmazonS3Trigger.spec.ts:53]: with role its not working

File: cypress/integration/72-triggers/AmazonS3Trigger.spec.ts (hunk 1 above)
Comment (satyam-harness) [reply cypress/integration/72-triggers/AmazonS3Trigger.spec.ts:56]: addressed

**Review by udayvunnam — APPROVED** [review]

File: cypress/fixtures/pipeline/api/triggers/amazonS3PipelineDetailsWithFilePathFixed.json [hunk 5]
```diff
@@ -0,0 +1,10 @@
+{
```
Comment (chetanvekariya-harness) [inline cypress/fixtures/pipeline/api/triggers/amazonS3PipelineDetailsWithFilePathFixed.json:1]: filePathRegexFixed ?

File: cypress/fixtures/pipeline/api/triggers/amazonS3pipelineDetailsWithRoutingIdCallWithFilePathFixed.json [hunk 6]
```diff
@@ -0,0 +1,11 @@
+{
```
Comment (chetanvekariya-harness) [inline cypress/fixtures/pipeline/api/triggers/amazonS3pipelineDetailsWithRoutingIdCallWithFilePathFixed.json:1]: FilePathRegexFixed ?

File: cypress/integration/72-triggers/AmazonS3Trigger.spec.ts [hunk 7]
```diff
@@ -0,0 +1,109 @@
+import {
+  inputSetsTemplateCall,
+  pipelineDetails,
+  pipelineDetailsWithRoutingIdCall,
+  pipelineStudioRoute,
+  s3bucketData,
+  triggersList
+} from '../../support/70-pipeline/constants'
+
+describe.skip('AmazonS3 Trigger', () => {
```
Comment (chetanvekariya-harness) [inline cypress/integration/72-triggers/AmazonS3Trigger.spec.ts:10]: Why is it skipped ?

File: cypress/integration/72-triggers/AmazonS3Trigger.spec.ts (hunk 1 above)
Comment (chetanvekariya-harness) [reply cypress/integration/72-triggers/AmazonS3Trigger.spec.ts:56]: Is this pushed ? Can't see the change ?

File: cypress/integration/72-triggers/AmazonS3Trigger.spec.ts [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+    cy.contains('span', '+ New Trigger').click()
+    cy.get('[data-cy="Artifact_Amazon S3"]').click()
+    cy.findByPlaceholderText('Enter Name').clear().type('amazonS3Trigger')
+    cy.contains('p', '+ Select Artifact').click()
+    cy.get('.TableV2--cells').eq(0).click()
+    cy.contains('span', 'Select').click()
+    cy.wait(1000)
+    cy.findByPlaceholderText('Select or Add bucket name').click()
+    cy.contains('p', 'tdp-tdp1-vzx7j8k9faso').click()
+    cy.get('input[value="testRegex"]').should('be.visible')
```
Comment (chetanvekariya-harness) [inline cypress/integration/72-triggers/AmazonS3Trigger.spec.ts:98]: Add an assertion which check if this is disabled.

File: src/modules/70-pipeline/strings/strings.en.yaml [hunk 9]
```diff
@@ -879,6 +879,7 @@ ciCodebase:
 tagDependencyRequired: required to fetch the tags
 artifactPathDependencyRequired: required to fetch the artifact paths
 artifactRepositoryDependencyRequired: 'Artifactory Connector is required to fetch repositories. You can make this field Runtime input otherwise.'
+amazonS3DependencyRequired: 'AmazonS3 Connector is required to fetch buckets.'
```
Comment (chetanvekariya-harness) [inline src/modules/70-pipeline/strings/strings.en.yaml:882]: Just display `Connector is required to fetch buckets`

File: src/modules/72-triggers/pages/triggers/utils/TriggersWizardPageUtils.ts [hunk 10]
```diff
@@ -1299,6 +1299,19 @@ const getManifestTableItem = ({
...
     } else {
+      if (manifest.type === 'Jenkins') {
+        return (
+          getRuntimeInputLabel({ str: manifest?.spec?.build, getString }) !==
+          getString?.('pipeline.artifactTriggerConfigPanel.runtimeInput')
+        )
+      }
+      if (manifest.type === 'AmazonS3') {
+        return (
+          !manifest?.spec?.filePathRegex &&
```
Comment (chetanvekariya-harness) [inline src/modules/72-triggers/pages/triggers/utils/TriggersWizardPageUtils.ts:1310]: You should use `||` here
`!manifest?.spec?.filePathRegex ||
          getRuntimeInputLabel({ str: manifest?.spec?.filePathRegex, getString }) !==
            getString?.('pipeline.artifactTriggerConfigPanel.runtimeInput')
        )`

File: src/modules/72-triggers/pages/triggers/views/modals/SelectArtifactModal.tsx [hunk 11]
```diff
@@ -147,6 +147,10 @@ const onSubmit = ({
     finalArtifact.spec.tag = replaceTriggerDefaultBuild({
       build: finalArtifact?.spec?.tag
     })
+  } else if (!isManifest && finalArtifact?.spec?.build) {
+    finalArtifact.spec.build = replaceTriggerDefaultBuild({
```
Comment (chetanvekariya-harness) [inline src/modules/72-triggers/pages/triggers/views/modals/SelectArtifactModal.tsx:151]: Please check this again. Maybe, it should be like below -
`finalArtifact.spec.filePathRegex = replaceTriggerDefaultBuild({`

File: src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/ArtifactSource/AmazonS3ArtifactSource/AmazonS3ArtifactSource.tsx [hunk 12]
```diff
@@ -184,7 +189,16 @@ const Content = (props: AmazonS3ContentProps): JSX.Element => {
               label={getString('pipeline.manifestType.bucketName')}
               placeholder={getString('pipeline.manifestType.bucketPlaceHolder')}
               name={`${path}.artifacts.${artifactPath}.spec.bucketName`}
-              disabled={isFieldDisabled(`artifacts.${artifactPath}.spec.bucketName`, true)}
+              disabled={
+                fromTrigger
```
Comment (chetanvekariya-harness) [inline src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/ArtifactSource/AmazonS3ArtifactSource/AmazonS3ArtifactSource.tsx:193]: Let's not even call the function here, it is confusing
Do like this
`!fromTrigger && isFieldDisabled(`artifacts.${artifactPath}.spec.bucketName`, true)`

**Review by chetanvekariya-harness — CHANGES_REQUESTED** [review]

File: src/modules/72-triggers/pages/triggers/views/modals/SelectArtifactModal.tsx (hunk 11 above)
Comment (satyam-harness) [reply src/modules/72-triggers/pages/triggers/views/modals/SelectArtifactModal.tsx:151]: what i have written is correct, this change is for jenkins trigger, not s3

File: src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/ArtifactSource/AmazonS3ArtifactSource/AmazonS3ArtifactSource.tsx (hunk 12 above)
Comment (satyam-harness) [reply src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/ArtifactSource/AmazonS3ArtifactSource/AmazonS3ArtifactSource.tsx:193]: okay

File: cypress/integration/72-triggers/AmazonS3Trigger.spec.ts (hunk 7 above)
Comment (satyam-harness) [reply cypress/integration/72-triggers/AmazonS3Trigger.spec.ts:10]: will fail if not skipped, as triggers are hidden from UI

File: src/modules/70-pipeline/strings/strings.en.yaml (hunk 9 above)
Comment (satyam-harness) [reply src/modules/70-pipeline/strings/strings.en.yaml:882]: okay

File: src/modules/72-triggers/pages/triggers/utils/TriggersWizardPageUtils.ts (hunk 10 above)
Comment (satyam-harness) [reply src/modules/72-triggers/pages/triggers/utils/TriggersWizardPageUtils.ts:1310]: usecase -> when filePathRegex is runtime/fixed then we have to keep artifact enabled. so || will not work. But i checked we can skip 2nd condition as well now.

File: cypress/integration/72-triggers/AmazonS3Trigger.spec.ts (hunk 1 above)
Comment (satyam-harness) [reply cypress/integration/72-triggers/AmazonS3Trigger.spec.ts]: yes

File: cypress/fixtures/pipeline/api/triggers/amazonS3PipelineDetailsWithFilePathFixed.json (hunk 5 above)
Comment (satyam-harness) [reply cypress/fixtures/pipeline/api/triggers/amazonS3PipelineDetailsWithFilePathFixed.json]: done, thank you

File: cypress/fixtures/pipeline/api/triggers/amazonS3pipelineDetailsWithRoutingIdCallWithFilePathFixed.json (hunk 6 above)
Comment (satyam-harness) [reply cypress/fixtures/pipeline/api/triggers/amazonS3pipelineDetailsWithRoutingIdCallWithFilePathFixed.json]: done, thank you

File: cypress/integration/72-triggers/AmazonS3Trigger.spec.ts [hunk 13]
```diff
@@ -95,7 +95,7 @@ describe.skip('AmazonS3 Trigger', () => {
     cy.wait(1000)
     cy.findByPlaceholderText('Select or Add bucket name').click()
     cy.contains('p', 'tdp-tdp1-vzx7j8k9faso').click()
-    cy.get('input[value="testRegex"]').should('be.visible')
+    cy.get('input[disabled]').should('be.visible')
```
Comment (chetanvekariya-harness) [inline cypress/integration/72-triggers/AmazonS3Trigger.spec.ts:98]: Any other input with disabled state will match with this. Proper way to write this is finding input with name attr and then checking if it is disabled.

**Review by chetanvekariya-harness — APPROVED** [review]

**Comment by satyam-harness** [comment]
retrigger jest

**Comment by satyam-harness** [comment]
retrigger jest

**Review by udayvunnam — APPROVED** [review]

**Review by bhavya-sinha — APPROVED** [review]

**Pull request merged**